src/OpenTelemetry.AutoInstrumentation.Loader/StartupLogger.cs

```diff
@@ -132,10 +135,32 @@ public static void Debug(string message, params object[] args)
 
         try
         {
+            // Pick up the parts used to build the log file name and minimize the chances
+            // of file name conflict with other processes.
             using var process = Process.GetCurrentProcess();
-            var appDomainName = AppDomain.CurrentDomain.FriendlyName;
-            // Do our best to not block other processes on write
-            return Path.Combine(LogDirectory, $"otel-dotnet-auto-loader-{appDomainName}-{process.Id}.log");
+
+            // AppDomain friendly name can contain characters that are invalid in file names,
+            // remove any of those. For the first assembly loaded by the process this is typically
+            // expected to be name of the file with the application entry point.
+            var appDomainFriendlyName = AppDomain.CurrentDomain.FriendlyName;
+            var invalidChars = new string(Path.GetInvalidFileNameChars()) + new string(Path.GetInvalidPathChars());
```

Comment:
Should it not be `HashSet<char>` instead of `string` to make the lookup of `invalidChars` O(1)?

Reply:
I had a similar Q, but, without time to do the small benchmark required to make an informed decision, especially considering that it runs once per instrumented AppDomain. Anyway, you are right we should benchmark alternatives, tracking via #1997